### PR TITLE
Wrap ref resolution errors with context

### PR DESCRIPTION
**Pull request by negz**
### Description of your changes

See https://github.com/crossplane/crossplane/issues/1499

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: apis/ec2/v1alpha4/referencers.go [hunk 1]
```diff
@@ -39,12 +41,12 @@ func (mg *RouteTable) ResolveReferences(ctx context.Context, c client.Reader) er
 		Extract:      reference.ExternalName(),
 	})
 	if err != nil {
-		return err
+		return errors.Wrap(err, "spec.vpcId")
```
Comment (muvaf) [inline apis/ec2/v1alpha4/referencers.go:44]:
```suggestion
		return errors.Wrap(err, "spec.forProvider.vpcId")
```

File: apis/ec2/v1alpha4/referencers.go [hunk 2]
```diff
@@ -54,13 +56,13 @@ func (mg *RouteTable) ResolveReferences(ctx context.Context, c client.Reader) er
 			Extract:      reference.ExternalName(),
 		})
 		if err != nil {
-			return err
+			return errors.Wrapf(err, "spec.routes[%d].gatewayId", i)
```
Comment (muvaf) [inline apis/ec2/v1alpha4/referencers.go:59]:
```suggestion
			return errors.Wrapf(err, "spec.forProvider.routes[%d].gatewayId", i)
```

File: apis/ec2/v1alpha4/referencers.go [hunk 3]
```diff
@@ -39,12 +41,12 @@ func (mg *RouteTable) ResolveReferences(ctx context.Context, c client.Reader) er
...
 	})
 	if err != nil {
-		return err
+		return errors.Wrap(err, "spec.vpcId")
 	}
 	mg.Spec.ForProvider.VPCID = aws.String(rsp.ResolvedValue)
 	mg.Spec.ForProvider.VPCIDRef = rsp.ResolvedReference
 
-	// Resolve spec.routes[].gatewayID
+	// Resolve spec.routes[].gatewayId
```
Comment (muvaf) [inline apis/ec2/v1alpha4/referencers.go:49]:
```suggestion
	// Resolve spec.forProvider.routes[].gatewayId
```

File: apis/ec2/v1alpha4/referencers.go [hunk 4]
```diff
@@ -54,13 +56,13 @@ func (mg *RouteTable) ResolveReferences(ctx context.Context, c client.Reader) er
...
 		if err != nil {
-			return err
+			return errors.Wrapf(err, "spec.routes[%d].gatewayId", i)
 		}
 		mg.Spec.ForProvider.Routes[i].GatewayID = aws.String(rsp.ResolvedValue)
 		mg.Spec.ForProvider.Routes[i].GatewayIDRef = rsp.ResolvedReference
 	}
 
-	// Resolve spec.associations[].subnetID
+	// Resolve spec.associations[].subnetId
```
Comment (muvaf) [inline apis/ec2/v1alpha4/referencers.go:65]:
```suggestion
	// Resolve spec.forProvider.associations[].subnetId
```

File: apis/ec2/v1alpha4/referencers.go [hunk 5]
```diff
@@ -70,7 +72,7 @@ func (mg *RouteTable) ResolveReferences(ctx context.Context, c client.Reader) er
 			Extract:      reference.ExternalName(),
 		})
 		if err != nil {
-			return err
+			return errors.Wrapf(err, "spec.associations[%d].subnetId", i)
```
Comment (muvaf) [inline apis/ec2/v1alpha4/referencers.go:75]:
```suggestion
			return errors.Wrapf(err, "spec.forProvider.associations[%d].subnetId", i)
```

File: apis/ec2/v1beta1/referencers.go [hunk 6]
```diff
@@ -41,7 +42,7 @@ func SecurityGroupName() reference.ExtractValueFn {
 func (mg *InternetGateway) ResolveReferences(ctx context.Context, c client.Reader) error {
 	r := reference.NewAPIResolver(c, mg)
 
-	// Resolve spec.vpcID
+	// Resolve spec.vpcId
```
Comment (muvaf) [inline apis/ec2/v1beta1/referencers.go:45]:
```suggestion
	// Resolve spec.forProvider.vpcId
```

File: apis/ec2/v1beta1/referencers.go [hunk 7]
```diff
@@ -50,7 +51,7 @@ func (mg *InternetGateway) ResolveReferences(ctx context.Context, c client.Reade
 		Extract:      reference.ExternalName(),
 	})
 	if err != nil {
-		return err
+		return errors.Wrap(err, "spec.vpcId")
```
Comment (muvaf) [inline apis/ec2/v1beta1/referencers.go:54]:
```suggestion
		return errors.Wrap(err, "spec.forProvider.vpcId")
```

File: apis/ec2/v1beta1/referencers.go [hunk 8]
```diff
@@ -62,7 +63,7 @@ func (mg *InternetGateway) ResolveReferences(ctx context.Context, c client.Reade
 func (mg *SecurityGroup) ResolveReferences(ctx context.Context, c client.Reader) error {
 	r := reference.NewAPIResolver(c, mg)
 
-	// Resolve spec.vpcID
+	// Resolve spec.vpcId
```
Comment (muvaf) [inline apis/ec2/v1beta1/referencers.go:66]:
```suggestion
	// Resolve spec.forProvider.vpcId
```

File: apis/ec2/v1beta1/referencers.go [hunk 9]
```diff
@@ -71,7 +72,7 @@ func (mg *SecurityGroup) ResolveReferences(ctx context.Context, c client.Reader)
 		Extract:      reference.ExternalName(),
 	})
 	if err != nil {
-		return err
+		return errors.Wrap(err, "spec.vpcId")
```
Comment (muvaf) [inline apis/ec2/v1beta1/referencers.go:75]:
```suggestion
		return errors.Wrap(err, "spec.forProvider.vpcId")
```

File: apis/ec2/v1beta1/referencers.go [hunk 10]
```diff
@@ -83,7 +84,7 @@ func (mg *SecurityGroup) ResolveReferences(ctx context.Context, c client.Reader)
 func (mg *Subnet) ResolveReferences(ctx context.Context, c client.Reader) error {
 	r := reference.NewAPIResolver(c, mg)
 
-	// Resolve spec.vpcID
+	// Resolve spec.vpcId
```
Comment (muvaf) [inline apis/ec2/v1beta1/referencers.go:87]:
```suggestion
	// Resolve spec.forProvider.vpcId
```

File: apis/ec2/v1beta1/referencers.go [hunk 11]
```diff
@@ -92,7 +93,7 @@ func (mg *Subnet) ResolveReferences(ctx context.Context, c client.Reader) error
 		Extract:      reference.ExternalName(),
 	})
 	if err != nil {
-		return err
+		return errors.Wrap(err, "spec.vpcId")
```
Comment (muvaf) [inline apis/ec2/v1beta1/referencers.go:96]:
```suggestion
		return errors.Wrap(err, "spec.forProvider.vpcId")
```

File: apis/identity/v1beta1/referencers.go [hunk 12]
```diff
@@ -19,11 +19,14 @@ package v1beta1
...
 	"context"
 
+	"github.com/pkg/errors"
+
 	"github.com/crossplane/provider-aws/apis/identity/v1alpha1"
 
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
 	"github.com/crossplane/crossplane-runtime/pkg/reference"
 	"github.com/crossplane/crossplane-runtime/pkg/resource"
```
Comment (muvaf) [inline apis/identity/v1beta1/referencers.go:29]:
```suggestion
	"sigs.k8s.io/controller-runtime/pkg/client"

	"github.com/crossplane/crossplane-runtime/pkg/reference"
	"github.com/crossplane/crossplane-runtime/pkg/resource"

	"github.com/crossplane/provider-aws/apis/identity/v1alpha1"
```

File: apis/notification/v1alpha1/referencers.go [hunk 13]
```diff
@@ -19,15 +19,17 @@ package v1alpha1
...
+
+	"github.com/crossplane/crossplane-runtime/pkg/reference"
 )
 
 // ResolveReferences for SNS Subscription managed type
 func (mg *SNSSubscription) ResolveReferences(ctx context.Context, c client.Reader) error {
 	r := reference.NewAPIResolver(c, mg)
 
-	// Resolve spec.TopicARN
+	// Resolve spec.topicArn
```
Comment (muvaf) [inline apis/notification/v1alpha1/referencers.go:32]:
```suggestion
	// Resolve spec.forProvider.topicArn
```

File: apis/notification/v1alpha1/referencers.go [hunk 14]
```diff
@@ -37,7 +39,7 @@ func (mg *SNSSubscription) ResolveReferences(ctx context.Context, c client.Reade
 	})
 
 	if err != nil {
-		return err
+		return errors.Wrap(err, "spec.topicArn")
```
Comment (muvaf) [inline apis/notification/v1alpha1/referencers.go:42]:
```suggestion
		return errors.Wrap(err, "spec.forProvider.topicArn")
```

File: apis/route53/v1alpha1/referencers.go [hunk 15]
```diff
@@ -19,17 +19,19 @@ package v1alpha1
...
+
 	"github.com/crossplane/provider-aws/apis/ec2/v1beta1"
 )
 
 // ResolveReferences of this Zone
 func (mg *ResourceRecordSet) ResolveReferences(ctx context.Context, c client.Reader) error {
 	r := reference.NewAPIResolver(c, mg)
 
-	// Resolve spec.zoneID
+	// Resolve spec.zoneId
```
Comment (muvaf) [inline apis/route53/v1alpha1/referencers.go:34]:
```suggestion
	// Resolve spec.forProvider.zoneId
```

File: apis/route53/v1alpha1/referencers.go [hunk 16]
```diff
@@ -38,7 +40,7 @@ func (mg *ResourceRecordSet) ResolveReferences(ctx context.Context, c client.Rea
 		Extract:      reference.ExternalName(),
 	})
 	if err != nil {
-		return err
+		return errors.Wrap(err, "spec.zoneId")
```
Comment (muvaf) [inline apis/route53/v1alpha1/referencers.go:43]:
```suggestion
		return errors.Wrap(err, "spec.forProvider.zoneId")
```

File: apis/storage/v1alpha1/referencers.go [hunk 17]
```diff
@@ -37,7 +39,7 @@ func S3BucketIAMUser() reference.ExtractValueFn {
 // ResolveReferences of this S3BucketPolicy
 func (mg *S3BucketPolicy) ResolveReferences(ctx context.Context, c client.Reader) error {
 	r := reference.NewAPIResolver(c, mg)
-	// Resolve spec.BucketName
+	// Resolve spec.bucketName
```
Comment (muvaf) [inline apis/storage/v1alpha1/referencers.go:42]:
```suggestion
	// Resolve spec.policyBody.bucketName
```

File: apis/storage/v1alpha1/referencers.go [hunk 18]
```diff
@@ -46,12 +48,12 @@ func (mg *S3BucketPolicy) ResolveReferences(ctx context.Context, c client.Reader
 		Extract:      reference.ExternalName(),
 	})
 	if err != nil {
-		return err
+		return errors.Wrap(err, "spec.bucketName")
```
Comment (muvaf) [inline apis/storage/v1alpha1/referencers.go:51]:
```suggestion
		return errors.Wrap(err, "spec.policyBody.bucketName")
```

File: apis/storage/v1alpha1/referencers.go [hunk 19]
```diff
@@ -46,12 +48,12 @@ func (mg *S3BucketPolicy) ResolveReferences(ctx context.Context, c client.Reader
...
 	}
 	mg.Spec.PolicyBody.BucketName = reference.ToPtrValue(rsp.ResolvedValue)
 	mg.Spec.PolicyBody.BucketNameRef = rsp.ResolvedReference
 
-	// Resolve spec.UserName
+	// Resolve spec.userName
```
Comment (muvaf) [inline apis/storage/v1alpha1/referencers.go:56]:
```suggestion
	// Resolve spec.policyBody.userName
```

File: apis/storage/v1alpha1/referencers.go [hunk 20]
```diff
@@ -60,7 +62,7 @@ func (mg *S3BucketPolicy) ResolveReferences(ctx context.Context, c client.Reader
 		Extract:      S3BucketIAMUser(),
 	})
 	if err != nil {
-		return err
+		return errors.Wrap(err, "spec.userName")
```
Comment (muvaf) [inline apis/storage/v1alpha1/referencers.go:65]:
```suggestion
		return errors.Wrap(err, "spec.policyBody.userName")
```

**Comment by negz** [comment]
Yikes; seems like I copied from lots of stale comments. Thanks @muvaf - PTAL.

File: apis/acmpca/v1alpha1/referencers.go [hunk 21]
```diff
@@ -37,7 +38,7 @@ func (mg *CertificateAuthorityPermission) ResolveReferences(ctx context.Context,
 		Extract:      reference.ExternalName(),
 	})
 	if err != nil {
-		return err
+		return errors.Wrap(err, "spec.forProvider.CertificateAuthorityARN")
```
Comment (muvaf) [inline apis/acmpca/v1alpha1/referencers.go:41]:
```suggestion
		return errors.Wrap(err, "spec.forProvider.certificateAuthorityARN")
```

File: apis/storage/v1alpha1/referencers.go [hunk 22]
```diff
@@ -46,12 +48,12 @@ func (mg *S3BucketPolicy) ResolveReferences(ctx context.Context, c client.Reader
 		Extract:      reference.ExternalName(),
 	})
 	if err != nil {
-		return err
+		return errors.Wrap(err, "spec.forProvider.bucketName")
```
Comment (muvaf) [inline apis/storage/v1alpha1/referencers.go:51]: I think this needs a rebase. S3 v1beta1 is merged and bucket name is now in external-name annotation.

Comment (negz) [reply apis/storage/v1alpha1/referencers.go]: I think this one is working as expected (it's for the bucket policy), but I've rebased and updated the bucket ones too.

**Review by hasheddan — COMMENTED** [review]
@negz not sure if I am not seeing latest but it looks like some of the comments such as this one: https://github.com/crossplane/provider-aws/pull/366/files#r500064830 are still not resolved

**Review by hasheddan — APPROVED** [review]
LGTM!

**Pull request merged**